File: scratch/refactor.py
import re
import os

filepath = 'c:\\Users\\frn\\Documents\\cadiz.taxi\\src\\pages\\index.astro'

with open(filepath, 'r', encoding='utf-8') as f:
    content = f.read()

# Extract styles
style_match = re.search(r'<style is:global>(.*?)</style>', content, re.DOTALL)
if style_match:
    style_content = style_match.group(1)
    os.makedirs('c:\\Users\\frn\\Documents\\cadiz.taxi\\src\\styles', exist_ok=True)
    with open('c:\\Users\\frn\\Documents\\cadiz.taxi\\src\\styles\\chat.css', 'w', encoding='utf-8') as f:
        f.write(style_content)
    
    # Insert import into frontmatter
    # Find the end of frontmatter
    frontmatter_end = content.find('---', 3)
    if frontmatter_end != -1:
        new_frontmatter = content[:frontmatter_end] + "import '../styles/chat.css';\n"
        content = new_frontmatter + content[frontmatter_end:]
    
    # Remove the old style tag
    content = content.replace(style_match.group(0), "")

# Extract scripts
script_match = re.search(r'<script>(.*?)</script>', content, re.DOTALL)
if script_match:
    script_content = script_match.group(1)
    os.makedirs('c:\\Users\\frn\\Documents\\cadiz.taxi\\src\\scripts', exist_ok=True)
    with open('c:\\Users\\frn\\Documents\\cadiz.taxi\\src\\scripts\\chat-client.ts', 'w', encoding='utf-8') as f:
        f.write(script_content)
    content = content.replace(script_match.group(0), '<script src="../scripts/chat-client.ts"></script>')

with open(filepath, 'w', encoding='utf-8') as f:
    f.write(content)

print("Refactored successfully")

File: scratch/refactor.mjs
import fs from 'fs';
import path from 'path';

const filepath = 'c:\\Users\\frn\\Documents\\cadiz.taxi\\src\\pages\\index.astro';
let content = fs.readFileSync(filepath, 'utf-8');

// Extract styles
const styleRegex = /<style is:global>([\s\S]*?)<\/style>/;
const styleMatch = content.match(styleRegex);
if (styleMatch) {
    const styleContent = styleMatch[1];
    fs.mkdirSync('c:\\Users\\frn\\Documents\\cadiz.taxi\\src\\styles', { recursive: true });
    fs.writeFileSync('c:\\Users\\frn\\Documents\\cadiz.taxi\\src\\styles\\chat.css', styleContent, 'utf-8');
    
    const frontmatterEnd = content.indexOf('---', 3);
    if (frontmatterEnd !== -1) {
        const newFrontmatter = content.substring(0, frontmatterEnd) + "import '../styles/chat.css';\n";
        content = newFrontmatter + content.substring(frontmatterEnd);
    }
    
    content = content.replace(styleMatch[0], '');
}

// Extract scripts
const scriptRegex = /<script>([\s\S]*?)<\/script>/;
const scriptMatch = content.match(scriptRegex);
if (scriptMatch) {
    const scriptContent = scriptMatch[1];
    fs.mkdirSync('c:\\Users\\frn\\Documents\\cadiz.taxi\\src\\scripts', { recursive: true });
    fs.writeFileSync('c:\\Users\\frn\\Documents\\cadiz.taxi\\src\\scripts\\chat-client.ts', scriptContent, 'utf-8');
    content = content.replace(scriptMatch[0], '<script src="../scripts/chat-client.ts"></script>');
}

fs.writeFileSync(filepath, content, 'utf-8');
console.log("Refactored successfully");
